### ISSUE TITLE
Support Actors for Process Starters

### ISSUE THREAD

**Issue by StephenOTT** [kiegroup/kogito-runtimes#1896]
### Description

Continuation of https://github.com/kiegroup/kogito-apps/issues/1200

The ProcessResourceGenerator.java supports the securityRoles.  Would you accept a PR for adding Actor support so Process Start security mirrors that of User Task security?

### Implementation ideas

The ProcessResourceGenerator and related locations that implement the securityRoles check will have a few lines added to support an actors check so specific users can be supported.

**Comment by StephenOTT** [comment]
@cristianonicolai  yes roles in practice are the best practice and SHOULD be used over usernames.  In real world thought, there are many times where you need to code in usernames for dev, quick fixes, "just get it done", etc use cases.  Same concepts to why UTs have Who the task is claimed to, the candidate Actors and candidate groups rather than just groups.

If we are going to create groups as the standard, then we need a ability to easily create sudo-groups that can be created through app config or somewhere else.  Something as simple as: "i am building a new process, and need a group created, but i can't create the group in my ODIC yet" becomes a blocker to dev without some faux/internal/app-memory-group ability.

**Comment by cristianonicolai** [comment]
@StephenOTT yep, I totally get that. The point is that I think there are specific libs/frameworks that can help that better than us creating something. Of course, we can create dev specific helpers, if that is somehow too much work, but see this file-based properties from Quarkus as an example: https://quarkus.io/guides/security-properties that can be easily enabled for dev, as long as we rely on standard auth.

**Comment by StephenOTT** [comment]
@cristianonicolai can the sec properties be used in **parallel** with ODIC/openid auth?

**Comment by StephenOTT** [comment]
But profiles would mean different ENVs.  In my scenario it is all the same ENV configurations: Process Starters that are coded to specific usernames would be at a dev level, but also at a production level.

If only roles are being used, are you planning to modify how user tasks function and only support "groups" / roles?
